code/modules/mob/living/basic/basic_defense.dm

```diff
@@ -31,6 +31,11 @@
 						"<span class='notice'>[user] [response_help_continuous] you.</span>", null, null, list(user))
 		to_chat(user, "<span class='notice'>You [response_help_simple] [src].</span>")
 		playsound(loc, 'sound/weapons/thudswoosh.ogg', 50, TRUE, -1)
+
+		// Vampires gain humanity from petting
+		if(IS_VAMPIRE(user))
+			var/datum/antagonist/vampire/vampdatum = IS_VAMPIRE(user)
+			SEND_SIGNAL(vampdatum, COMSIG_VAMPIRE_TRACK_HUMANITY_GAIN, HUMANITY_PETTING_TYPE, src)
```

Comment:
This is bad code, specific vampire overrides in a generic function. Find a better way to do this